src/merge.rs

```diff
@@ -45,6 +71,70 @@ impl MergeValue {
                 hasher.write_byte(*zero_count);
                 hasher.finish()
             }
+            #[cfg(feature = "trie")]
+            MergeValue::ShortCut {
+                key: _,
+                value,
+                height,
+            } => {
+                // try keep hash same with MergeWithZero
+                if value.is_zero() {
+                    return H256::zero();
+                }
+                if *height == 0 {
+                    *value
+                } else {
+                    self.into_merge_with_zero::<H>().hash::<H>()
+                }
+            }
+        }
+    }
+
+    /// Helper function for get base_node
+    /// When call with MergeValue::Value(v), it would be v
+    pub fn base_node<H: Hasher + Default>(&self) -> H256 {
+        match self {
+            #[cfg(feature = "trie")]
+            MergeValue::ShortCut {
+                key,
+                value,
+                height: _,
+            } => {
+                let base_key = key.parent_path(0);
+                hash_base_node::<H>(0, &base_key, value)
+            }
+            MergeValue::MergeWithZero {
+                base_node,
+                zero_bits: _,
+                zero_count: _,
+            } => *base_node,
+            MergeValue::Value(value) => *value,
+        }
+    }
+
+    /// Helper function for Shortcut node
+    /// Transform it into a MergeWithZero node
+    #[cfg(feature = "trie")]
```

Comment:
Since this is a public function, should mention when this will panic.

https://doc.rust-lang.org/src/core/iter/traits/iterator.rs.html#234-237